proposals/0000-template.md

```diff
@@ -83,6 +83,36 @@ learnability of the language for novice users. Define and list any remaining
 drawbacks that cannot be resolved.
 
 
+## Backward Compatibility
+
+Will your proposed change cause any existing programs to change behaviour or
+stop working? Assess the expected impact on existing code on the following scale:
+
+0. No breakage
```

Comment:
Good questions. My answers would be:
 * If a proposal adds an extension, and turning on that extension breaks code, that isn't a backwards compatibility problem as such. Existing code will still compile, because it won't have the extension enabled. It might be a design consideration that influences the desirability of the extension, of course.
 * Changes to existing extensions are considered breaking, regardless of whether the extension has been advertised as "unstable", though we might be more willing to accept breaking changes to such features. We don't currently have a clear demarcation between "stable" and "unstable" extensions, and this is to some extent a matter of community consensus.

My intention is that whether something is a "breaking change" should be objective and correspond closely to whether code that compiled with GHC version X still compiles with version X+1. But acceptance of proposals is about weighing various factors, of which potential breakage should be only one.

Reply:
We probably want to know about both kinds of breakage: even if it's guarded by an extension the breakage becomes relevant when we consider turning the extension on by default, down the road, and so it may be relevant how much breakage to expect then (unless the extension is meant to stay a flag forever)

Reply:
Fair point. I can try to change the proposal text to reflect this.